router/errors.go

```diff
@@ -8,16 +8,18 @@ import (
 	"github.com/bugsnag/bugsnag-go/v2"
 	"github.com/netlify/netlify-commons/metriks"
 	"github.com/netlify/netlify-commons/tracing"
+	"github.com/sirupsen/logrus"
 )
 
 // HTTPError is an error with a message and an HTTP status code.
 type HTTPError struct {
-	Code            int         `json:"code"`
-	Message         string      `json:"msg"`
-	JSON            interface{} `json:"json,omitempty"`
-	InternalError   error       `json:"-"`
-	InternalMessage string      `json:"-"`
-	ErrorID         string      `json:"error_id,omitempty"`
+	Code            int             `json:"code"`
+	Message         string          `json:"msg"`
+	JSON            interface{}     `json:"json,omitempty"`
+	InternalError   error           `json:"-"`
+	InternalMessage string          `json:"-"`
+	ErrorID         string          `json:"error_id,omitempty"`
+	Fields          []logrus.Fields `json:"-"`
```

Comment:
`logrus.Fields` is just an alias for a hashmap. multiple entries should be merged, which is the same logrus does for multiple calls to `WithFields`
you could also use a `*logrus.Entry` which you could call `WithFields` on to use their merging logic

Reply:
I added some naive merging logic, and a test: https://github.com/netlify/netlify-commons/pull/326/commits/b692cb1e06019e100b57897652c251e3c4627ff3